pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -302,7 +316,59 @@ func (c *Controller) onServiceEvent(curr interface{}, event model.Event) error {
 }
 
 func (c *Controller) onNodeEvent(obj interface{}, ev model.Event) error {
-	return c.checkReadyForEvents()
+	if err := c.checkReadyForEvents(); err != nil {
+		return err
+	}
+
+	var k8sNodes []*kubernetesNode
+	var extaddresses []string
+	rawNodes := c.nodes.GetStore().List()
+	for _, i := range rawNodes {
+		machine := i.(*v1.Node)
+		for _, address := range machine.Status.Addresses {
+			if address.Type == v1.NodeExternalIP {
+				k8sNodes = append(k8sNodes,
+					&kubernetesNode{address: address.Address,
+						labels: labels.Collection{machine.Labels, machine.Annotations}})
```

Comment:
Why need this?

Reply:
So this will come in future. In infrastrufctures with 100+ nodes for a k8s cluster, people dont want a gateway traffic to land in some arbitrary node. They have some firewall zones. They have setup pod placement policies to ensure that the gateway is scheduled only on a subset of nodes. In this case, they will want to tell us how to select these specific nodes for the gateway. The label selector is one way of achieving the goal.